tests/test-team/template-mgmt-component-tests/template-mgmt-choose-page.component.ts

```diff
@@ -130,42 +167,106 @@ test.describe('Choose Template Type Page', () => {
+  for (const { label, path } of [
+    { label: 'NHS App message', path: 'nhs-app' },
+    { label: 'Email', path: 'email' },
+    { label: 'Text message (SMS)', path: 'text-message' },
+    { label: 'Letter', path: 'letter' },
+  ])
+    test(`should navigate to the ${label} template creation page when radio button selected and continue button clicked`, async ({
+      page,
+      baseURL,
+    }) => {
+      const chooseTemplatePage = new TemplateMgmtChoosePage(page);
+
+      await chooseTemplatePage.navigateToChooseTemplatePage(
+        sessionDataForRadioSelect.id
+      );
+      await TemplateMgmtChoosePage.checkRadioButton(page.getByLabel(label));
+      await chooseTemplatePage.clickContinueButton();
+
+      await expect(page).toHaveURL(
+        `${baseURL}/templates/create-${path}-template/${sessionDataForRadioSelect.id}`
+      );
+    });
+
+  for (const { label, sessionData } of [
+    { label: 'NHS App message', sessionData: nhsAppRadioSelectedSessionData },
+    { label: 'Email', sessionData: emailRadioSelectedSessionData },
+    { label: 'Text message (SMS)', sessionData: smsRadioSelectedSessionData },
+    { label: 'Letter', sessionData: letterRadioSelectedSessionData },
+  ]) {
+    test(`should display ${label} radio button selected if present in session storage`, async ({
+      page,
+    }) => {
+      const chooseTemplatePage = new TemplateMgmtChoosePage(page);
+
+      await chooseTemplatePage.navigateToChooseTemplatePage(sessionData.id);
+
+      expect(page.getByLabel(label)).toBeChecked();
+    });
+  }
+
+  test('should not display "Go Back" link on page', async ({ page }) => {
     const chooseTemplatePage = new TemplateMgmtChoosePage(page);
 
-    await chooseTemplatePage.navigateToChooseTemplatePage(
-      emptySessionDataForRadioSelect.id
-    );
-    await TemplateMgmtChoosePage.checkRadioButton(
-      page.locator('[id="templateType-NHS_APP"]')
-    );
-    await chooseTemplatePage.clickContinueButton();
+    await chooseTemplatePage.navigateToChooseTemplatePage(emptySessionData.id);
 
-    await expect(page).toHaveURL(
-      `${baseURL}/templates/create-nhs-app-template/${emptySessionDataForRadioSelect.id}`
-    );
+    await expect(chooseTemplatePage.goBackLink).toBeHidden();
   });
 
-  test('should display NHS App radio button selected if present in session storage', async ({
+  test('should navigate to error page if sessionId invalid', async ({
     page,
+    baseURL,
   }) => {
     const chooseTemplatePage = new TemplateMgmtChoosePage(page);
 
-    await chooseTemplatePage.navigateToChooseTemplatePage(
-      nhsAppRadioSelectedSessionData.id
-    );
+    await chooseTemplatePage.navigateToChooseTemplatePage('invalid-sessionId');
 
-    // expect to assert radio selected
+    await expect(page).toHaveURL(`${baseURL}/templates/invalid-session`);
   });
 
-  test('should not display "Go Back" link on page', async ({ page }) => {
+  test('Footer links exist and are visible', async ({ page }) => {
```

Comment:
not sure what this test is adding - this is just testing the static appearance of the page which is covered by unit tests

Reply:
Agree, although can be built on in future to test the linkiness of the links i guess. Again copied and pasted, think this will change when rebased with Bens changes aswell

Reply:
removed and replaced with common test step for footer links, refactored tests to check for href attribute over them being visible